DigitalOffice/Services/TimeManagementService/Validators/CreateWorkTimeRequestValidator.cs

```diff
@@ -0,0 +1,93 @@
+        /// </summary>
+        public readonly static int toDay = 2;
+        /// <summary>
+        /// Limit on working hours in a row.
+        /// </summary>
+        public readonly static TimeSpan workingLimit = new TimeSpan(24, 0, 0); // 24 hours.
+
+        private readonly DateTime fromDateTime = DateTime.Now.AddDays(-fromDay);
+        private readonly DateTime toDateTime = DateTime.Now.AddDays(toDay);
+
+        public CreateWorkTimeRequestValidator([FromServices] IWorkTimeRepository repository)
+        {
+            this.repository = repository;
+
+            RuleFor(wt => wt.WorkerUserId)
+                    .NotEmpty();
+
+            RuleFor(wt => wt.Date)
+                .NotEmpty()
+                .When(wt => wt.Date > fromDateTime)
+                .WithMessage(wt => string.Format(
+                    "WorkTime had to be filled no later than {0}.",
+                    fromDateTime.ToString(System.Globalization.CultureInfo.CreateSpecificCulture("en-GB"))))
+                .When(wt => wt.Date < toDateTime)
+                .WithMessage(wt => string.Format(
+                    "WorkTime cannot be filled until {0}.",
+                    fromDateTime.ToString(System.Globalization.CultureInfo.CreateSpecificCulture("en-GB"))));
+
+            RuleFor(wt => wt.Time)
+                .NotEmpty()
+                .When(wt => wt.Time.TotalMinutes <= 0)
+                .WithMessage("You cannot indicate that you worked zero hours or a negative amount.")
+                .When(wt => Math.Round(wt.Time.TotalMinutes, 12) > Math.Round(workingLimit.TotalMinutes, 12))
+                .WithMessage(wt => string.Format(
+                    "You cannot indicate that you worked more than {0} hours and {1} minutes.",
+                    workingLimit.Hours, workingLimit.Minutes))
+                .When(wt =>
+                {
+                    var startTimeOfNewWT = wt.Date;
+                    var endTimeOfNewWT = startTimeOfNewWT.AddMinutes(wt.Time.TotalMinutes);
+
+                    var workTimes = repository.GetUserWorkTimes(
+                        wt.WorkerUserId,
+                        new WorkTimeFilter
+                        {
+                            StartTime = startTimeOfNewWT.AddMinutes(-workingLimit.TotalMinutes),
+                            EndTime = endTimeOfNewWT.AddMinutes(workingLimit.TotalMinutes)
+                        });
+
+                    foreach (var oldWT in workTimes)
+                    {
+                        var startDate = oldWT.Date;
+                        var endDate = oldWT.Date.AddMinutes(oldWT.Time.TotalMinutes);
+
+                        if (endTimeOfNewWT <= startDate) { } // First new WorkTime, then old workTime
+                        else if (endDate <= startTimeOfNewWT) { } // First old WorkTime, then new WorkTime
+                        else
+                        {
+                            return false;
+                        }
+                    }
```

Comment:
It might be a good idea to move this check into a separate static method. You can see an example in the documentation.
https://docs.fluentvalidation.net/en/latest/custom-validators.html
We need to refactor the code. So far, it's very hard to figure it out.

Reply:
Also, you use When, but should use Must. Check documentation, please.

Reply:
> Also, you use When, but should use Must. Check documentation, please.

Fixed.